packages/context-menu/src/vaadin-contextmenu-items-mixin.js

```diff
@@ -71,14 +71,23 @@ export const ItemsMixin = (superClass) =>
       };
     }
 
+    /**
+     * Tag name prefix used by overlay, list-box and items.
+     * @protected
+     * @return {string}
+     */
+    get _tagNamePrefix() {
+      return 'vaadin-context-menu';
+    }
+
     /** @protected */
     ready() {
       super.ready();
 
       // Overlay's outside click listener doesn't work with modeless
       // overlays (submenus) so we need additional logic for it
       this.__itemsOutsideClickListener = (e) => {
-        if (!e.composedPath().some((el) => el.localName === 'vaadin-context-menu-overlay')) {
+        if (!e.composedPath().some((el) => el.localName === `${this._tagNamePrefix}-overlay`)) {
```

Comment:
Would `this.constructor.is` be too restrictive for this?

Reply:
In case of `vaadin-menu-bar` we have `vaadin-menu-bar-submenu` and that would mean the following:

- `vaadin-menu-bar-submenu-overlay`
- `vaadin-menu-bar-submenu-list-box`
- `vaadin-menu-bar-submenu-item`

These tag names seem to bee too complicated to me, so I decided not to use `submenu` for them.